Add nepali numerals eval

samyok opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, **failure to follow the guidelines below will result in the PR being closed automatically**. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access be granted. 🚨

**PLEASE READ THIS**:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject it since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

BAlso, please note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available 
[here](https://git-lfs.com).

## Eval details 📑

### Eval name

nepali-numerals

### Eval description

Test the model's ability to convert Nepali numerals to numbers.

### What makes this a useful eval?

Ensure the model understands numbers when written in Nepali.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can create an eval on 
cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the 
`Criteria` Model-graded eval.
- [x] **Include at least 15 high-quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should

- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your YAML is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (<https://platform.openai.com/docs/usage-policies>).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the commits on the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgment

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and the high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access be granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
{"input": [{"role": "system", "content": "You will be prompted with a Nepali cardinal numeral, representing an integer number between 0 and 999999999. Your task is to convert this numeral to its respective number. Don't use any delimiters, output only the digits and nothing else."}, {"role": "user", "content": "एक"}], "ideal": "1"}
{"input": [{"role": "system", "content": "You will be prompted with a Nepali cardinal numeral, representing an integer number between 0 and 999999999. Your task is to convert this numeral to its respective number. Don't use any delimiters, output only the digits and nothing else."}, {"role": "user", "content": "तीन"}], "ideal": "3"}
{"input": [{"role": "system", "content": "You will be prompted with a Nepali cardinal numeral, representing an integer number between 0 and 999999999. Your task is to convert this numeral to its respective number. Don't use any delimiters, output only the digits and nothing else."}, {"role": "user", "content": "दश"}], "ideal": "10"}
{"input": [{"role": "system", "content": "You will be prompted with a Nepali cardinal numeral, representing an integer number between 0 and 999999999. Your task is to convert this numeral to its respective number. Don't use any delimiters, output only the digits and nothing else."}, {"role": "user", "content": "अठइस"}], "ideal": "28"}
{"input": [{"role": "system", "content": "You will be prompted with a Nepali cardinal numeral, representing an integer number between 0 and 999999999. Your task is to convert this numeral to its respective number. Don't use any delimiters, output only the digits and nothing else."}, {"role": "user", "content": "अत्दै गएका"}], "ideal": "29"}
{"input": [{"role": "system", "content": "You will be prompted with a Nepali cardinal numeral, representing an integer number between 0 and 999999999. Your task is to convert this numeral to its respective number. Don't use any delimiters, output only the digits and nothing else."}, {"role": "user", "content": "पैतिस"}], "ideal": "35"}
{"input": [{"role": "system", "content": "You will be prompted with a Nepali cardinal numeral, representing an integer number between 0 and 999999999. Your task is to convert this numeral to its respective number. Don't use any delimiters, output only the digits and nothing else."}, {"role": "user", "content": "चौवालिस"}], "ideal": "44"}
  ```
</details>

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request a change.

The evaluation method being used to evaluate the answer is `Match` which only works when an exact match is found. The user needs to provide very specific instructions to the model about the output format, and the ideal answer should be in that expected format. In this dataset, the model is being instructed to convert Nepali numerals to numbers. There are no specific instructions about the output format, and the model may provide a number like `100,000`, but the ideal answer expects an output like `100000`. Due to this, the `Match` evaluation will mark the completion wrong despite the right answer being provided by the model.

You either need to add proper prompt instructions to tell the model about the expected format of output or you need to modify the format of your ideal answer.

I also need some clarification regarding some samples. Although I'm not familiar with Nepali numerals or Nepali script, I'm familiar with Hindi and Urdu numerals, and after listening to these numerals, it seems like there is some issue with the answers. For example, consider this sample:

`सात सय साठम्स दुई हजार चार सय साठी`

The ideal answer is `4737176`, but it seems like the provided numeral should end with `460`.

Similarly, for the numeral `आठ सय रतरसत्तर तीन हजार र बीस`, the ideal answer is `8658592`, but it seems like this number should be ended with `3020`.

I'm not familiar with the Nepali numerals, so I might be wrong. Can you provide me with clarification on these?

We would love to review the PR again after the suggested changes.

usama-openai commented:
For sample `छ सय र तीस आठ हजार र NINT`, what is role of `NINT` here? Is it by mistake or is it part of the number?

samyok commented:
You're right. For some reason, when creating the evals, I copied the _delta_ of the model's answers from the actual. I'll clean that up, my bad.

samyok commented:
Hey @usama-openai !! You're right, I did have some data issues. I cleaned up the data significantly and added more prompting to ensure that no commas or spaces fall through the cracks.

Thanks so much!!

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.

usama-openai commented:
You should see GPT-4 API access enabled in your account in the next few days.
